3rdparty/CMakeLists.txt

```diff
@@ -578,8 +583,17 @@ function(NBL_ADJUST_FOLDER NBL_TARGET NBL_PREFIX KEEP_PRESENT_PREFIX)
 	endif()
 endfunction()
 
-NBL_GET_ALL_TARGETS(NBL_ALL_TARGETS)
-foreach(NBL_TARGET IN LISTS NBL_ALL_TARGETS)
+NBL_GET_ALL_TARGETS(NBL_ALL_3RDPARTY_TARGETS)
+foreach(NBL_TARGET IN LISTS NBL_ALL_3RDPARTY_TARGETS)
+	get_target_property(NBL_TARGET_SOURCE_DIR ${NBL_TARGET} SOURCE_DIR)
+	if(${NBL_TARGET_SOURCE_DIR} MATCHES "${NBL_ROOT_PATH}/3rdparty/dxc*" OR ${NBL_TARGET_SOURCE_DIR} MATCHES "${NBL_ROOT_PATH}/3rdparty/libjpeg-turbo*")
+		message("found:${NBL_TARGET}:${NBL_TARGET_SOURCE_DIR}")
+		list(APPEND NBL_IMMUTABLE_TARGETS NBL_TARGET)
+	endif()
+endforeach()
+
+NBL_GET_ALL_TARGETS(NBL_ALL_3RDPARTY_TARGETS)
```

Comment:
don't use this util second time here, at this point we don't add any new 3rdparty targets so you do the same job twice, we should call it only once just before updating their properties/compile options, definitions or whatever